File: modules/library-loader/src/main/java/team/idealstate/hyper/library/loader/api/entity/Repository.java
/*
 *    hyper-library-loader
 *    Copyright [2024] [ideal-state]
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package team.idealstate.hyper.library.loader.api.entity;

import java.io.Serializable;
import java.net.URI;
import java.util.Objects;

/**
 * <p>Repository</p>
 *
 * <p>创建于 2024/3/28 11:03</p>
 *
 * @author ketikai
 * @version 1.0.0
 * @since 1.0.0
 */
public class Repository implements Serializable {
    private static final long serialVersionUID = 3390425019236889369L;

    private String name;
    private URI url;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public URI getUrl() {
        return url;
    }

    public void setUrl(URI url) {
        this.url = url;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Repository)) {
            return false;
        }
        final Repository that = (Repository) o;

        if (!Objects.equals(name, that.name)) {
            return false;
        }
        return Objects.equals(url, that.url);
    }

    @Override
    public int hashCode() {
        int result = name != null ? name.hashCode() : 0;
        result = 31 * result + (url != null ? url.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Repository{" +
                "name='" + name + '\'' +
                ", url=" + url +
                '}';
    }
}

File: buildSrc/src/main/kotlin/team/idealstate/hyper/gradle/module/repository/entity/Repository.kt
/*
 *    hyper-gradle-plugin
 *    Copyright [2024] [ideal-state]
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package team.idealstate.hyper.gradle.module.repository.entity

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.deser.std.StdDeserializer
import java.net.URI

/**
 * <p>Repository</p>
 *
 * <p>创建于 2024/3/20 17:01</p>
 * @author ketikai
 * @since 1.0.0
 * @version 1.0.0
 */
data class Repository(
    @JsonDeserialize(using = Type.Deserializer::class)
    val type: Type,
    val name: String,
    val url: URI,
    val username: String?,
    val password: String?,
) {
    enum class Type {
        MAVEN,
        IVY;

        internal object Deserializer : StdDeserializer<Type>(Type::class.java) {
            private const val serialVersionUID: Long = 4371258286182194375L

            override fun deserialize(parser: JsonParser, context: DeserializationContext): Type {
                return valueOf(parser.text.uppercase())
            }
        }
    }
}